### DISCUSSION TITLE
Create authenctication flow that lets a user select active role in client after sign in?

### DISCUSSION THREAD

**Discussion by agreedSkiing** [keycloak/keycloak#15164] [Q&A] [answered]
Is it possible to create an authenticaiton flow after a user has signed in to a client that lets them (the user) choose thier active role in the client and store that active role in the jwt token?

**Comment by dominikschlosser** [comment, marked as answer]
Hi, we have a similar requirement. I created a proof-of-concept here: https://github.com/dominikschlosser/keycloak-extension-playground/tree/main/extensions/role-filter

It consists of a custom Authenticator and ProtocolMapper and should be straight forward to understand.

**Reply by agreedSkiing** [reply]
Okey, so it's not possible to do it without an extension good to know and the code is straight forward to understand :)
